linetools/isgm/utils.py

```diff
@@ -463,53 +471,65 @@ def group_coincident_compoments(comp_list, output_type='list'):
     if output_type not in ['list', 'dict', 'dictionary']:
         raise ValueError("`output_type` must be either 'list' or 'dict'.")
 
-    # the first extreme case is that all components are independent
-    # of each other, in which case we have the following output shape
-    out = [[] for kk in range(len(comp_list))]
-
-    for ii in range(len(comp_list)):
-        comp_ii = comp_list[ii]
-        # only append if ii does not belong to a previous round
-        switch = 0
-        for kk in range(len(out[:ii])):
-            if ii in out[kk]:
-                switch = 1
-                break
-        if switch == 1:
-            pass
+    ### We first want to identify and group all blended lines
+    ### Sort them by observed wavelength to do this
+    lst=[]
+    compnos=[]
+    for ii,comp in enumerate(comp_list):
+        lst.extend(comp._abslines)
+        compnos.extend([ii]*len(comp._abslines))
+    lst=np.array(lst)
+    compnos=np.array(compnos)
+    wv1s=np.array([line.limits.wvlim[0].value for line in lst])
+    sortidxs=np.argsort(wv1s)
+    sort_lst=lst[sortidxs]
+    sort_compnos=compnos[sortidxs] # This will store indices of the lines' parent comps.
+
+    ### Identify the blends
+    blends = []
+    for i in range(len(sortidxs)-1):
+        if i == 0:
+            thisblend = [i]
+        if sort_lst[i].coincident_line(sort_lst[i+1]):
+            thisblend.append(i+1)
+            if i==(len(sortidxs)-2):
+            	blends.append(thisblend)
         else:
-            out[ii].append(ii)
```

Comment:
Could you keep my old method rather than delete it? You can do this by renaming it like `group_coincident_compoments_old`. In this way we can investigate how the two implementations compare in terms of results/speed/etc.